### ISSUE TITLE
`multiple-statements` no longer detected in a `finally:` block

### ISSUE THREAD

**Issue by oprypin** [pylint-dev/pylint#9759]
### Bug description

```python
"""hi"""

try:
    pass
finally: pass  # Should complain about [multiple-statements] but doesn't
```

### Configuration

```ini
# Empty configuration
```

### Command used

```shell
/tmp/pylint/venv/bin/pylint test.py
```

### Pylint output

```
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

```
************* Module test
test.py:5:9: C0321: More than one statement on a single line (multiple-statements)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)
```

This behavior used to happen in the past.

5bdc2218ef9891e9047823c26625c62d68f2dc52 is the first bad commit

### Pylint version

```shell
pylint 3.3.0-dev0
astroid 3.3.0-dev0
Python 3.11.8 (main, May 26 2024, 00:40:16) [GCC 13.2.0]
```

**Comment by nickdrozd** [comment]
`try` and `except` are counted, while `else` and `finally` are not:

```python
try:     pass   # flagged
except:  pass   # flagged
else:    pass   # ignored
finally: pass   # ignored
```
